### PR TITLE
Resolved minor change concerning the Fan consisting of GPIO Enable

### PR DESCRIPTION
**Pull request by pro-sper**

### CODE REVIEW

File: include/TMS/dev/RadiatorFan.hpp [hunk 1]
```diff
@@ -35,6 +35,8 @@ class RadiatorFan {
 private:
     /** PWM instance to control the fan */
     IO::PWM& pwm;
+    /** GPIO instance to turn on the fan */
+    IO::GPIO& gpio = IO::getGPIO<IO::Pin::PC_1>();
```
Comment (mjmagee991) [inline include/TMS/dev/RadiatorFan.hpp]: This should be passed as an input to the constructor

File: src/TMS/dev/RadiatorFan.cpp [hunk 2]
```diff
@@ -8,10 +8,16 @@ RadiatorFan::RadiatorFan(IO::PWM& pwm): pwm(pwm) {
 }
 
     void RadiatorFan::setSpeed(uint16_t speed) {
-        if (speed > MAX_SPEED) {
-            speed = MAX_SPEED;
+        if (speed == 0){
+            gpio.writePin(EVT::core::IO::GPIO::State::LOW);
         }
-        pwm.setDutyCycle(speed);
+        else{
```
Comment (mjmagee991) [inline src/TMS/dev/RadiatorFan.cpp:14]: It looks like the autoformatter didn't run on this branch. Make sure it does before you merge

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Minor implementation issue

**Review by AarC10 — APPROVED** [review]

File: targets/fanSample/main.cpp [hunk 3]
```diff
@@ -23,6 +23,8 @@ int main() {
 
     // Simple code to test the fan at different speeds
 
+
```
Comment (ishan-renjen) [inline targets/fanSample/main.cpp:27]: remove line

File: include/TMS/dev/RadiatorFan.hpp [hunk 4]
```diff
@@ -35,6 +35,8 @@ class RadiatorFan {
 private:
     /** PWM instance to control the fan */
     IO::PWM& pwm;
+    /** GPIO instance to turn on the fan */
```
Comment (ishan-renjen) [inline include/TMS/dev/RadiatorFan.hpp:38]: add a newline  between 37 and 38

**Review by ishan-renjen — APPROVED** [review]

**Review by S-TH3TA-13 — APPROVED** [review]

**Review by mjmagee991 — APPROVED** [review]

**Pull request merged**